saleor/warehouse/availability.py

```diff
@@ -94,3 +125,49 @@ def is_product_in_stock(
         country_code, channel_slug, product
     ).annotate_available_quantity()
     return any(stocks.values_list("available_quantity", flat=True))
+
+
+def get_reserved_quantity(
+    stocks: StockQuerySet, current_checkout_lines: Optional[List["CheckoutLine"]] = None
+) -> int:
+    result = (
+        Reservation.objects.filter(
+            stock__in=stocks,
+        )
+        .not_expired()
+        .exclude_checkout_lines(current_checkout_lines)
+        .aggregate(
+            quantity_reserved=Coalesce(Sum("quantity_reserved"), 0),
+        )
+    )  # type: ignore
+
+    return result["quantity_reserved"]
+
+
+def get_reserved_quantity_bulk(
+    stocks: Iterable[Stock],
+    current_checkout_lines: Optional[List["CheckoutLine"]] = None,
+) -> Dict[int, int]:
+    reservations: Dict[int, int] = defaultdict(int)
+    if not stocks:
+        return reservations
+
+    result = (
+        Reservation.objects.filter(
```

Comment:
it is only a suggestion, but you could have some helper function that will return `result` structure. You could use it here and inside `get_reserved_quantity`